File: lesson_5_2/app/src/main/java/com/kalashnyk/denys/samplemvvmapp/domain/AllUsersViewModel.java
package com.kalashnyk.denys.samplemvvmapp.domain;

import android.annotation.SuppressLint;
import android.app.Application;
import android.arch.lifecycle.LiveData;
import android.util.Log;

import com.kalashnyk.denys.samplemvvmapp.presentation.widget.SingleLiveEvent;
import com.kalashnyk.denys.samplemvvmapp.repository.AppRepository;
import com.kalashnyk.denys.samplemvvmapp.repository.database.entity.UserEntity;

import java.util.List;

public class AllUsersViewModel extends BaseViewModel {

    private AppRepository mRepository;
    private SingleLiveEvent<List<UserEntity>> liveDataItems = new SingleLiveEvent<List<UserEntity>>();

    public AllUsersViewModel(Application application, AppRepository repository) {
        super(application);
        mRepository = repository;
    }

    @SuppressLint("CheckResult")
    public void getAllItems() {
        mRepository.getAllUsers().subscribe(list -> liveDataItems.setValue(list));
    }

    public LiveData<List<UserEntity>> getLiveDataItems() {
        return liveDataItems;
    }
}

File: lesson_7/app/src/main/java/com/kalashnyk/denys/kotlinsample/domain/AllUsersViewModel.kt
package com.kalashnyk.denys.kotlinsample.domain

import android.annotation.SuppressLint
import android.app.Application
import android.arch.lifecycle.LiveData
import com.kalashnyk.denys.kotlinsample.presentation.widget.SingleLiveEvent
import com.kalashnyk.denys.kotlinsample.repository.AppRepository
import com.kalashnyk.denys.kotlinsample.repository.database.entity.UserEntity

class AllUsersViewModel(application: Application, private val mRepository: AppRepository) : BaseViewModel(application) {
    private val liveDataItems = SingleLiveEvent<List<UserEntity>>()

    @SuppressLint("CheckResult")
    fun getAllItems() {
        mRepository.getAll()?.subscribe { list -> liveDataItems.value = list }
    }

    fun getLiveDataItems(): LiveData<List<UserEntity>> {
        return liveDataItems
    }
}

